### DISCUSSION TITLE
crashonly feature junit enhancement

### DISCUSSION THREAD

**Discussion by andrlos** [junit-team/junit-framework#3692] [General]
Hi!
it has been a few months since I started attempting implementing crashonly enhancement into [jtreg/jtharness](https://bugs.openjdk.org/browse/CODETOOLS-7903519) repositories and failed in the end.
The goal of this enhancement is creation of an option that would allow tester to skip regular test failures and only report jvm crashes. As one of the people testing jdk builds this would greatly benefit us in our fastdebug testing.
There is a number of ways to detect a jvm crash and it ultimately only comes down to where a person is allowed to detect it.
The easiest way of such a detection would be to keep an eye of what execution of individual tests produced and if it produced hs_err_pid.log file which is a symptom of a crash.
This however might not be such a foolproof way since even the creation of this log can fail. The optimal solution would be to both monitor creation of this log and monitor exit codes of the system probably. But this is open to discussion.
Hence I would love to ask the community three questions:
1. Would you allow an attempt to implement this into junit?
2. If yes, where would you see this going in the code?
3. Do you have any propositions on how to achieve this? I might have missed something obvious and there might be a mechanism allowing this.
Thanks for reading!

**Comment by sormuras** [comment]
Hi @andrlos - I guess, this feature needs to implemented outside of the tools and libraries you mentioned; but with the help of those tools.

Does this scratch show your use-case?

```java
void main() {
  var code = exec("jtreg", ... "test/SomeTest.java");
  if (code != 0 || Files.exists(Path.of("hs_err_pid.log"))) {
    // print helpful error analysis messages
    System.exit(1);
  }
  System.exit(0);
}
```
